can-observe.js

```diff
@@ -1,35 +1,211 @@
+		var old = [].slice.call(this, 0);
+		var args = Array.from(arguments);
+		var ret = protoFn.apply(this, arguments);
+		canReflect.eachKey(changeEvents, function(makeArgs, event) {
+			var allArgs = makeArgs(this, args, ret, old);
+			allArgs.forEach(function(handlerArgs) {
+				canEvent.dispatch.call(this, event, handlerArgs);
+			}, this);
+		}.bind(this));
+		handlers.length.forEach(function(handler){
+			canBatch.queue([handler, this, [this.length, old.length]]);
+		}, this);
+		return ret;
+	};
+});
+
+var observe = function(obj){
+	if (obj[observableSymbol]) {
+		return obj[observableSymbol].proxy;
+	} else {
+		Object.defineProperty(obj, "_cid", {
+			value: cid({}),
+			enumerable: false
+		});
+	}
+	var isArray = obj instanceof Array;
 
-module.exports = function(obj){
-  cid(obj);
-	obj.addEventListener = canEvent.addEventListener;
-	obj.removeEventListener = canEvent.removeEventListener;
 	var p = new Proxy(obj, {
-		get: function(target, name){
-			if(name !== "_cid" && has.call(target, name)) {
-        		Observation.add(target, name);
+		get: function(target, key, receiver){
+			if(proxyOnly[key]) {
+				return proxyOnly[key];
+			}
+			var descriptor = Object.getOwnPropertyDescriptor(target, key);
+			var value;
+			if(descriptor && descriptor.get) {
+				value = descriptor.get.call(receiver);
+			} else {
+				if(key === "__bindEvents") {
+					value = target[observableSymbol].__bindEvents;
+				} else {
+					value = target[key];				
+				}
 			}
-			return target[name];
+			if (!canReflect.isSymbolLike(key) && !canReflect.isObservableLike(value) && canReflect.isPlainObject(value)) {
+				value = target[key] = observe(value);
+			}
+			if (isArray && typeof value === "function" && arrayMethodInterceptors[key]) {
+				value = arrayMethodInterceptors[key];
+			}
+			if (key !== "_cid" && key !== "__bindEvents" &&
+				typeof value !== "function" &&
+				!canReflect.isSymbolLike(key) && 
+				(hasOwn.call(target, key) || !Object.isSealed(target))
+			) {
+				Observation.add(receiver, key.toString());
+			}
+			return value;
 		},
-		set: function(target, key, value){
-			var old = target[key];
-			var change = old !== value;
-			if(change) {
-				target[key] = value;
-				canBatch.dispatch.call(target, {
-					type: key,
-					target: target
-				}, [value, old]);
+		set: function(target, key, value, receiver){
+			var old, change;
+			var descriptor = Object.getOwnPropertyDescriptor(target, key);
+			if (!canReflect.isSymbolLike(key) && !canReflect.isObservableLike(value) && canReflect.isPlainObject(value)) {
+				value = observe(value);
+			} else if (value && value[observableSymbol]){
+				value = value[observableSymbol].proxy;
+			}
+
+			if(descriptor && descriptor.set) {
+				descriptor.set.call(receiver, value);
+			} else {
+				old = target[key];
+				change = old !== value;
+				if (change) {
+					target[key] = value;
+				}
+			}
+			if(descriptor && descriptor.set || change) {
```

Comment:
why fire if there's a setter?

Reply:
Artifact from when getters and setters were observable.  Removed.